File: CSVExplorer/app/src/main/java/com/example/csvexplorer/entity/RowEntity.kt
package com.example.csvexplorer.entity


import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "rows")
data class RowEntity(
    @PrimaryKey(autoGenerate = true) val localId: Long = 0L,
    val externalId: String?,
    val dataJson: String
)

File: csv_explorer_flutter/lib/data/row_entity.dart
class RowEntity {
  final int? localId;
  final String? externalId;
  final String dataJson;

  RowEntity({this.localId, this.externalId, required this.dataJson});

  Map<String, Object?> toMap() => {
    'local_id': localId,
    'external_id': externalId,
    'data_json': dataJson,
  };

  static RowEntity fromMap(Map<String, Object?> m) => RowEntity(
    localId: m['local_id'] as int?,
    externalId: m['external_id'] as String?,
    dataJson: (m['data_json'] as String?) ?? '{}',
  );
}
